### PR TITLE
updated error reporting to display error and include warnings

### PR DESCRIPTION
**Pull request by natjoe4**
## Summary
Updated the return values from calculator functions to display errors with more information and include warnings.
# New Behavior
- Input that will cause fqm-execution to throw an error will now output an error message to the user and log the stack trace.
- Input that previously would have caused fqm-execution to print with console.warn will now return an additional warnings attribute containing messages with more detailed information.
- updated to fqm-execution 0.8.0
# Code Changes
- changed return objects and logging behavior for every function in app.ts
- updated tests to reflect new output format `(results) -> {results: results, warnings: withErrors}`
# Testing Guidance
- Run `npm run test` which has been updated to reflect the new output format
- Run `npm run start`, then using Insomnia, postman or your requests frontend of choice (I had success with [insomnia](https://insomnia.rest/pricing)) use the following urls to send POST requests (assuming your server is running on localhost:3000) : 
-- http://localhost:3000/calculate
-- http://localhost:3000/calculateRaw
-- http://localhost:3000/calculateMeasureReports
-- http://localhost:3000/Measure/$care-gaps
-- http://localhost:3000/Measure/$data-requirements
pass in JSON in the following format:
`{
"measure": *measure bundle contents*,
"patients": [*patient bundle contents*],
"options": {} (or add options if you'd like, but an empty object will work fine)
}`
--use the ColorectalCancerScreeningCQM-Bundle.json with the EXM130 denominator patients, and scroll to the bottom to see the new warnings attribute
-- enter an empty patient bundle array to see new error output format
-- enter any other combination of mb/pb in your arsenal and ensure the output is the same as the output from fqm-execution
-- I recommend pasting measure and patient bundles in and deleting attributes to invalidate the JSON, then running it to ensure the error received makes sense for the damage done. For example, removing the valueset URL from a measure bundle should throw a bad request error with the following output `{
  "name": "UnexpectedResource",
  "message": "Missing the following valuesets: http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.108.12.1038, and no API key was provided to resolve them"
}`

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -20,9 +20,20 @@ app.post('/calculateRaw', async (req, res) => {
...
     );
-    return res.json(rawResult.results);
+    /**
+     * TODO:
+     * Format of RawResults output should maybe be changed in fqm-execution to adapt to new withErrors field.
+     * Currently rawResults output is either the rawResults object (on a successful run) or an error message
+     * with an explanation for the failure. I think we should instead just throw an error upon failure, not
+     * return an error string.
+     */
+    return res.json({ results: rawResult.results, warnings: rawResult.withErrors });
```
Comment (natjoe4) [inline src/app.ts:30]: Please checkout above comment. lmk if you disagree!

Comment (sarahmcdougall) [reply src/app.ts:30]: I agree

File: test/app.test.ts [hunk 2]
```diff
@@ -115,5 +116,9 @@ test('simple calculate', async () => {
     .send({ measure: mockMeasureBundle, patients: mockPatientBundles })
     .expect(200);
 
-  expect(response.body).toEqual(mockResult.results);
+  expect(response.body.results).toEqual(mockResult.results);
 });
+function parseBundle(filePath: string): R4.IBundle {
```
Comment (sarahmcdougall) [inline test/app.test.ts:121]: Is the `parseBundle()` function needed in this test file? I'm seeing a warning that it's defined but never used

Comment (natjoe4) [reply test/app.test.ts:121]: Good point. I may end up needing it for the second half of this task to test the error/warning reporting, but makes sense to remove it for now.

**Review by sarahmcdougall — APPROVED** [review]
Functionality and syntax both look good to me. All the tests work as expected. Good work!

**Review by sarahmcdougall — APPROVED** [review]
Re-tested with the changes, and both functionality and syntax are good. All tests work as expected, just like before. Not sure if we need another reviewer on this, but if not then this is good to merge!

**Pull request merged**